### PR TITLE
feat(node-mono-repo): add release-please-config post-processing

### PR DESCRIPTION
**Pull request by sofisl**

### CODE REVIEW

File: synthtool/languages/node_mono_repo.py [hunk 1]
```diff
@@ -59,6 +59,18 @@ def read_metadata(relative_dir: str):
...
+def write_release_please_config(owlbot_dirs):
+    with open("release-please-config.json", "r+") as f:
+        data = json.load(f)
+        for dir in owlbot_dirs:
+            result = re.search(r"(packages/.*)", dir)
+            assert result is not None
+            data["packages"][result.group()] = {}
+        f.seek(0)
+        json.dump(data, f, indent=4)
+        f.truncate()
```
Comment (chingor13) [inline synthtool/languages/node_mono_repo.py:71]: Readability-wise, I would separate the read from the write. So open the file to read, then close. Then reopen the file to overwrite.

File: synthtool/languages/node_mono_repo.py [hunk 2]
```diff
@@ -59,6 +59,18 @@ def read_metadata(relative_dir: str):
...
 
+def write_release_please_config(owlbot_dirs):
+    with open("release-please-config.json", "r+") as f:
+        data = json.load(f)
+        for dir in owlbot_dirs:
+            result = re.search(r"(packages/.*)", dir)
+            assert result is not None
+            data["packages"][result.group()] = {}
+        f.seek(0)
+        json.dump(data, f, indent=4)
```
Comment (chingor13) [inline synthtool/languages/node_mono_repo.py:70]: Personally, I prefer 2 spaces :laughing:

File: tests/test_node_mono_repo.py [hunk 3]
```diff
@@ -139,6 +139,32 @@ def test_generate_index_ts():
         assert filecmp.cmp(generated_index_path, sample_index_path)
 
 
+def test_write_release_please_config():
+    # use a non-nodejs template directory
+    with util.chdir(FIXTURES / "node_templates" / "release_please"):
```
Comment (chingor13) [inline tests/test_node_mono_repo.py:144]: I would use `util.copied_fixtures_dir` instead. It copies the fixtures to a temporary directory and runs in that temporary directory context. This way, it won't modify the fixture in place.

File: tests/fixtures/node_templates/release_please/release-please-config-post.json [hunk 4]
```diff
@@ -0,0 +1,9 @@
+{
+    "release-type": "node",
```
Comment (chingor13) [inline tests/fixtures/node_templates/release_please/release-please-config-post.json:2]: This indentation does not match the other fixture. Let's pick a single one.

File: tests/test_node_mono_repo.py [hunk 5]
```diff
@@ -139,6 +139,32 @@ def test_generate_index_ts():
...
+        node_mono_repo.write_release_please_config(
+            [
+                "google-cloud-node/packages/gapic-node-templating",
+                "Users/person/google-cloud-node/packages/dlp",
+                "Users/person/google-cloud-node/packages/asset",
```
Comment (chingor13) [inline tests/test_node_mono_repo.py:149]: These seem like odd inputs, are they valid?

**Review by chingor13 — CHANGES_REQUESTED** [review]

**Comment by SurferJeffAtGoogle** [comment]
Please remove me as reviewer.  I'd remove myself, but I don't have permission.

File: tests/test_node_mono_repo.py (hunk 5 above)
Comment (sofisl) [reply tests/test_node_mono_repo.py:149]: I specifically created some strange inputs to test the regex.

**Comment by sofisl** [comment]
@SurferJeffAtGoogle, seems like I can't either, they're grayed out for me:

Screen Shot 2022-08-02 at 2 25 04 PM

File: synthtool/languages/node_mono_repo.py [hunk 6]
```diff
@@ -59,6 +59,19 @@ def read_metadata(relative_dir: str):
...
+    with open("release-please-config.json", "r") as f:
+        data = json.load(f)
+        for dir in owlbot_dirs:
+            result = re.search(r"(packages/.*)", dir)
+            assert result is not None
+            data["packages"][result.group()] = {}
+    with open("release-please-config.json", "w") as f:
+        f.seek(0)
+        json.dump(data, f, indent=2)
+        f.truncate()
```
Comment (chingor13) [inline synthtool/languages/node_mono_repo.py:72]: seek and truncate are no longer needed.

**Review by chingor13 — CHANGES_REQUESTED** [review]

**Review by chingor13 — DISMISSED** [review]

**Review by chingor13 — APPROVED** [review]

**Pull request merged**